sn_transfers/src/transfers/signed_transaction.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::collections::BTreeSet;
+
+use crate::error::Result;
+use crate::{
+    CashNote, DerivationIndex, MainPubkey, MainSecretKey, NanoTokens, OutputPurpose, SignedSpend,
+    SpendReason, TransferError, UnsignedTransaction,
+};
+use serde::{Deserialize, Serialize};
+
+/// A local transaction that has been signed and is ready to be executed on the Network
+#[derive(custom_debug::Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct SignedTransaction {
```

Comment:
do we really need the entire `trasaction` to be signed ?
I think we only need a bunch of `SignedTrasnfer`s (one generated per output_cashnotes) to be generated ?

Reply:
The transaction itself is not signed, but the `SignedSpend` contained in it are so we can finish the CashNote and as you say send them as `Transfers` which don't need to be signed as they contain the network addresses of the `SignedSpend`s

Reply:
Ah, ok.
just double checked the code of `let signed_tx = unsigned_tx.sign(main_key)?;` 
which the `UnSignedTransaction::sign` function only only sign the contains spends individually, instead of signing the entire body.
Though the API is little bit misleading (as the `sign` function normally indicates signing the entire body), the current processing logic is correct.